mlbench_core/utils/pytorch/utils.py

```diff
@@ -0,0 +1,16 @@
+import torch
+
+
+@torch.jit.script
+def orthogonalize(matrix):
+    """Function used to orthogonalize a matrix."""
+    n, m = matrix.shape
+    for i in range(m):
+        # Normalize the i'th column
+        col = matrix[:, i : i + 1]
+        col /= torch.sqrt(torch.sum(col ** 2))
```

Comment:
This could lead to division by zero. I'd suggest to add a `+ eps` here, which is configurable as a second argument and has a default of `eps=1e-16`. I'm not so familiar with the project yet, but if there are unit tests, would be good to add some here (columns have unit length & columns have zero dot product)

Reply:
Sure, I will add this. FYI, I copied the function from [here](https://github.com/epfml/powersgd/blob/3f4eb8d097e9a25ba4fd4b2ff1b2bf83c5aa665a/gradient_reducers.py#L945), so maybe it should be changed there too.